File: exercicios/implementacoes_java/trabalho_arvore/arvore_binaria/NoArvore.java
package arvore_binaria;

public class NoArvore{
    private NoArvore filhoEsquerdo;
    private NoArvore filhoDireito;
    private NoArvore pai;
    private int elemento;

    public NoArvore(NoArvore filhoEsquerdo, NoArvore filhoDireito, NoArvore pai, int o){
        this.filhoEsquerdo = filhoEsquerdo;
        this.filhoDireito = filhoDireito;
        this.pai = pai;
        this.elemento = o;
    }

    public NoArvore getFilhoEsquerdo(){
        return this.filhoEsquerdo;
    }

    public NoArvore getFilhoDireito(){
        return this.filhoDireito;
    }

    public NoArvore getPai(){
        return this.pai;
    }

    public int getElemento(){
        return this.elemento;
    }

    public void setFilhoEsquerdo(NoArvore v){
        this.filhoEsquerdo = v;
    }

    public void setFilhoDireito(NoArvore v){
        this.filhoDireito = v;
    }

    public void setPai(NoArvore v){
        this.pai = v;
    }

    public void setElemento(int o){
        this.elemento = o;
    }
}

File: exercicios/implementacoes_csharp/trabalho_arvore/binaria/NoArvore.cs
namespace ArvoreBinaria;

public class NoArvore {
    private NoArvore FilhoEsquerdo;
    private NoArvore FilhoDireito;
    private NoArvore Pai;
    private int Elemento;

    public NoArvore(NoArvore FilhoEsquerdo, NoArvore FilhoDireito, NoArvore Pai, int Elemento) {
        this.FilhoEsquerdo = FilhoEsquerdo;
        this.FilhoDireito = FilhoDireito;
        this.Pai = Pai;
        this.Elemento = Elemento;
    }

    public NoArvore GetFilhoEsquerdo() {
        return FilhoEsquerdo;
    }

    public NoArvore GetFilhoDireito() {
        return FilhoDireito;
    }

    public NoArvore GetPai() {
        return Pai;
    }

    public int GetElemento() {
        return Elemento;
    }

    public void SetFilhoEsquerdo(NoArvore No) {
        FilhoEsquerdo = No;
    }

    public void SetFilhoDireito(NoArvore No) {
        FilhoDireito = No;
    }

    public void SetPai(NoArvore No) {
        Pai = No;
    }

    public void SetElemento(NoArvore No) {
        Elemento = No;
    }
}